subprojects/diagnostics/src/integTest/groovy/org/gradle/api/tasks/diagnostics/ProjectReportTaskIntegrationTest.groovy

```diff
@@ -199,7 +203,9 @@ No sub-projects
                 public abstract UtilityExtension getUtility();
 
                 @Override
-                public void apply(Project project) {}
+                public void apply(Project project) {
+                    project.getExtensions().add("utility", getUtility());
+                }
```

Comment:
Could we add an example, in this test or elsewhere, of **not** registering the `@SoftwareType` as a project extension and it thus **not** being usable (and not showing up in these reports)?

It would also be interesting to see a `@SoftwareType` that uses the same name as an existing top-level block (say `configurations`) now not adding the project extension and thus being a useable name after this change.